astropy/io/fits/tests/test_image.py

```diff
@@ -73,7 +73,7 @@ def test_constructor_copies_header(self):
         # Original header should be unchanged
         assert phdr['FILENAME'] == 'labq01i3q_rawtag.fits'
 
-    @raises(ValueError)
+    @raises(IndexError)
```

Comment:
The only "unexpected" change is this: with the new behavior the index error when accessing the HDU happens before the `ValueError` that was raised when trying to access the data. So I think it's fine.